process/block/shardblock.go

```diff
@@ -1847,7 +1738,72 @@ func (sp *shardProcessor) getMaxMiniBlocksSpaceRemained(
 ) int32 {
 	mbSpaceRemainedInBlock := int32(maxItemsInBlock) - int32(itemsAddedInBlock)
 	mbSpaceRemainedInCache := int32(core.MaxMiniBlocksInBlock) - int32(miniBlocksAddedInBlock)
-	maxMbSpaceRemained := core.Min(mbSpaceRemainedInBlock, mbSpaceRemainedInCache)
+	maxMbSpaceRemained := core.MinInt32(mbSpaceRemainedInBlock, mbSpaceRemainedInCache)
 
 	return maxMbSpaceRemained
 }
+
+func (sp *shardProcessor) CreateBlockStarted() {
+	sp.txCoordinator.CreateBlockStarted()
+
+	sp.hdrsForCurrBlock.mutHdrsForBlock.Lock()
+	sp.hdrsForCurrBlock.missingHdrs = 0
+	sp.hdrsForCurrBlock.missingFinalHdrs = 0
+	sp.hdrsForCurrBlock.hdrHashAndInfo = make(map[string]*hdrInfo)
+	sp.hdrsForCurrBlock.mutHdrsForBlock.Unlock()
+}
+
+func (sp *shardProcessor) sortHdrsForCurrentBlock(usedInBlock bool) ([]*block.MetaBlock, error) {
+	hdrsForCurrentBlock := make([]*block.MetaBlock, 0)
+
+	sp.hdrsForCurrBlock.mutHdrsForBlock.RLock()
+	for _, hdrInfo := range sp.hdrsForCurrBlock.hdrHashAndInfo {
+		if hdrInfo.usedInBlock != usedInBlock {
+			continue
+		}
+
+		metaHdr, ok := hdrInfo.hdr.(*block.MetaBlock)
+		if !ok {
+			sp.hdrsForCurrBlock.mutHdrsForBlock.RUnlock()
+			return nil, process.ErrWrongTypeAssertion
+		}
+
+		hdrsForCurrentBlock = append(hdrsForCurrentBlock, metaHdr)
+	}
+	sp.hdrsForCurrBlock.mutHdrsForBlock.RUnlock()
+
+	if len(hdrsForCurrentBlock) > 1 {
+		sort.Slice(hdrsForCurrentBlock, func(i, j int) bool {
+			return hdrsForCurrentBlock[i].Nonce < hdrsForCurrentBlock[j].Nonce
+		})
+	}
+
+	return hdrsForCurrentBlock, nil
+}
+
+func (sp *shardProcessor) sortHdrsHashesForCurrentBlock(usedInBlock bool) [][]byte {
```

Comment:
todo: remove bool parameter and give instead the set to sort

Also the sort is done by nonce, so maybe it would be good to have this in the name.

Reply:
Done